lib/app_profiler.rb

```diff
@@ -116,6 +116,25 @@ def backend=(new_backend)
       @profiler_backend = new_profiler_backend
     end
 
+    def profile_sampler_enabled=(value)
+      if value.is_a?(Proc)
+        raise ArgumentError,
+          "profile_sampler_enabled must be a proc or a lambda that accepts no argument" if value.arity != 0
+      else
+        raise ArgumentError, "Must be TrueClass or FalseClass" unless [TrueClass, FalseClass].include?(value.class)
+      end
+
+      @profile_sampler_enabled = value
+    end
+
+    def profile_sampler_enabled
+      return false unless defined?(@profile_sampler_enabled)
```

Comment:
```suggestion
```

You don't need that. If it's not defined then it's `nil`. If it's `nil` it isn't a `Proc`. So you just return `nil` which is close enough to `false`.

Reply:
If it's okay, I would like to keep it as is. We enforce `True/False/Proc` when setting. So if we read `nil` and then try to set it back, it raises - which does not seem right.